Random failure of PEC sampling test

The test for `test_sample_sequence_choi` failed in the PEC module for python 3.7 in ubuntu in #461 as follows 
```
        ideal_op = gate.on(*qreg)
        noisy_op_tree = [ideal_op] + [depolarize(BASE_NOISE)(q) for q in qreg]
        ideal_choi = _operation_to_choi(gate.on(*qreg))
        noisy_choi = _operation_to_choi(noisy_op_tree)
        choi_unbiased_estimates = []
        for _ in range(500):
            imp_seq, sign, norm = sample_sequence(gate.on(*qreg), DECO_DICT)
            # Apply noise after each sequence.
            # NOTE: noise is not applied after each operation.
            noisy_sequence = [imp_seq] + [depolarize(BASE_NOISE)(q) for q in qreg]
            sequence_choi = _operation_to_choi(noisy_sequence)
            choi_unbiased_estimates.append(norm * sign * sequence_choi)
        choi_pec_estimate = np.average(choi_unbiased_estimates, axis=0)
        noise_error = np.linalg.norm(ideal_choi - noisy_choi)
        pec_error = np.linalg.norm(ideal_choi - choi_pec_estimate)
>       assert pec_error < noise_error
E       assert 0.02458828 < 0.023094043

mitiq/pec/tests/test_pec_sampling.py:162: AssertionError

----------- coverage: platform linux, python 3.7.9-final-0 -----------
Name                                          Stmts   Miss  Cover
-----------------------------------------------------------------
mitiq/__init__.py                                 4      0   100%
mitiq/_about.py                                  25     11    56%
mitiq/_typing.py                                  9      0   100%
mitiq/_version.py                                 5      0   100%
mitiq/benchmarks/__init__.py                      0      0   100%
mitiq/benchmarks/maxcut.py                       46      2    96%
mitiq/benchmarks/random_circuits.py              45      2    96%
mitiq/benchmarks/randomized_benchmarking.py      22      1    95%
mitiq/benchmarks/utils.py                        11      0   100%
mitiq/conversions.py                             52      1    98%
mitiq/mitiq_pyquil/__init__.py                    1      0   100%
mitiq/mitiq_pyquil/compiler.py                  149      0   100%
mitiq/mitiq_pyquil/conversions.py                17      0   100%
mitiq/mitiq_pyquil/pyquil_utils.py               29      1    97%
mitiq/mitiq_qiskit/__init__.py                    2      0   100%
mitiq/mitiq_qiskit/conversions.py                24      0   100%
mitiq/mitiq_qiskit/qiskit_utils.py               18      0   100%
mitiq/pec/__init__.py                             0      0   100%
mitiq/pec/decomposition/__init__.py               0      0   100%
mitiq/pec/decomposition/depolarizing.py          25      0   100%
mitiq/pec/pec.py                                 33      0   100%
mitiq/pec/sampling.py                            31      0   100%
mitiq/pec/types/__init__.py                       1      0   100%
mitiq/pec/types/types.py                        179      4    98%
mitiq/pec/utils.py                               79      0   100%
mitiq/utils.py                                   69      0   100%
mitiq/zne/__init__.py                             1      0   100%
mitiq/zne/inference.py                          407     25    94%
mitiq/zne/scaling/__init__.py                     3      0   100%
mitiq/zne/scaling/folding.py                    213      7    97%
mitiq/zne/scaling/parameter.py                   34      0   100%
mitiq/zne/zne.py                                 29      1    97%
-----------------------------------------------------------------
TOTAL                                          1563     55    96%
Coverage XML written to file coverage.xml

=========================== short test summary info ============================
FAILED mitiq/pec/tests/test_pec_sampling.py::test_sample_sequence_choi[gate0]
================== 1 failed, 531 passed in 606.16s (0:10:06) ===================
make: *** [test-all] Error 1
Makefile:70: recipe for target 'test-all' failed
Error: Process completed with exit code 2.
```
[Link](https://github.com/unitaryfund/mitiq/runs/1549193395).
It failed for only one ubuntu/python setup, and then did not faile again, it may be a random occurrence.

--- a/mitiq/pec/tests/test_pec_sampling.py
+++ b/mitiq/pec/tests/test_pec_sampling.py
@@ -149,8 +149,11 @@ def test_sample_sequence_choi(gate: Gate):
     ideal_choi = _operation_to_choi(gate.on(*qreg))
     noisy_choi = _operation_to_choi(noisy_op_tree)
     choi_unbiased_estimates = []
+    rng = np.random.RandomState(1)
     for _ in range(500):
-        imp_seq, sign, norm = sample_sequence(gate.on(*qreg), DECO_DICT)
+        imp_seq, sign, norm = sample_sequence(
+            gate.on(*qreg), DECO_DICT, random_state=rng
+        )
         # Apply noise after each sequence.
         # NOTE: noise is not applied after each operation.
         noisy_sequence = [imp_seq] + [depolarize(BASE_NOISE)(q) for q in qreg]

